pwndbg/aglib/dynamic.py

```diff
@@ -40,14 +37,7 @@ def _r_debug():
     [2]: https://elixir.bootlin.com/glibc/glibc-2.38/source/elf/dl-debug-symbols.S#L30
     """
 
-    try:
-        address = gdb.execute("output/x &_r_debug", to_string=True)
-        address = int(address, 0)
-
-        return address
-    except gdb.error:
-        # Symbol is most likely unavailable.
-        return None
+    return pwndbg.dbg.selected_inferior().symbol_address_from_name("_r_debug")
```

Comment:
As I said on Discord, we want to optimize this in the future so that one doesn't have to call selected inferior

Reply:
Maybe something like `pwndbg.dbg.inferior`? We always know when a new valid inferior gets created and when it gets destroyed, anyway, in both LLDB and GDB, so that wouldn't be much of an issue. Though the same can't be said for either `selected_thread()` or `selected_frame()` under LLDB. They might be harder to do correctly, given that sometimes - but not always - we have to pull them from the current inferior - see `_any_thread` and `_any_bottommost_frame`.